File: Week 05/id_437/70.爬楼梯.js
/*
 * @lc app=leetcode.cn id=70 lang=javascript
 *
 * [70] 爬楼梯
 */

// @lc code=start
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if (n === 1) return 1
    let m = []
    m[1] = 1
    m[2] =2
    for (let i = 3; i <= n; i++) {
      m[i] = m[i - 1] + m[i - 2]
    }
    return m[n]
};
// @lc code=end

File: Week 01/id_477/70.爬楼梯.go
package leetcode

/*
 * @lc app=leetcode.cn id=70 lang=golang
 *
 * [70] 爬楼梯
 */

// @lc code=start
func climbStairs(n int) int {
	if n <= 2 {
		return n
	}
	var s1, s2 = 1, 2
	for n > 2 {
		tmp := s1 + s2
		s1 = s2
		s2 = tmp
		n--
	}
	return s2
}

// @lc code=end
